File: src/main/java/org/springframework/samples/petclinic/rest/mapper/SpecialtyMapper.java
package org.springframework.samples.petclinic.rest.mapper;

import org.mapstruct.Mapper;
import org.springframework.samples.petclinic.vet.Specialty;
import org.springframework.samples.petclinic.rest.dto.SpecialtyDto;

@Mapper(componentModel = "spring")
public interface SpecialtyMapper {

	SpecialtyDto toDto(Specialty specialty);

}

File: src/main/kotlin/org/springframework/samples/petclinic/rest/mapper/SpecialtyMapper.kt
package org.springframework.samples.petclinic.rest.mapper

import org.springframework.samples.petclinic.rest.dto.SpecialtyDto
import org.springframework.samples.petclinic.vet.Specialty
import org.springframework.stereotype.Component

@Component
class SpecialtyMapper {

    fun toDto(specialty: Specialty): SpecialtyDto {
        return SpecialtyDto(
            id = specialty.id,
            name = specialty.name
        )
    }

}